File: app/src/main/java/com/android/ichooseyou/activity/TeamPickerActivity.java
package com.android.ichooseyou.activity;

import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.LinearLayout;
import android.widget.TextView;
import android.widget.Toast;
import android.widget.ImageView;

import androidx.appcompat.app.AppCompatActivity;
import androidx.core.content.ContextCompat;

import com.android.ichooseyou2.R;
import com.google.android.material.button.MaterialButton;
import com.google.android.material.card.MaterialCardView;
import com.google.android.material.textfield.TextInputEditText;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Random;

public class TeamPickerActivity extends AppCompatActivity {

    private TextInputEditText teamCountInput, participantsInput;
    private MaterialButton generateTeamsButton, saveTeamsButton, shuffleAgainButton;
    private LinearLayout teamsContainer;
    private TextView resultsTitle;
    private ArrayList<String> itemsList;
    private List<List<String>> generatedTeams = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_team_picker);

        // Initialize views
        teamCountInput = findViewById(R.id.team_count_input);
        participantsInput = findViewById(R.id.participants_input);
        generateTeamsButton = findViewById(R.id.generate_teams_button);
        saveTeamsButton = findViewById(R.id.save_teams_button);
        shuffleAgainButton = findViewById(R.id.shuffle_again_button);
        teamsContainer = findViewById(R.id.teams_container);
        resultsTitle = findViewById(R.id.results_title);

        // Get the items list from the intent
        itemsList = getIntent().getStringArrayListExtra("ITEMS");

        // If items were passed, populate the participants field
        if (itemsList != null && !itemsList.isEmpty()) {
            participantsInput.setText(TextUtils.join("\n", itemsList));
        }

        generateTeamsButton.setOnClickListener(v -> generateBalancedTeams());
        saveTeamsButton.setOnClickListener(v -> saveTeams());
        shuffleAgainButton.setOnClickListener(v -> shuffleTeams());
    }

    private void generateBalancedTeams() {
        // Validate inputs
        if (TextUtils.isEmpty(teamCountInput.getText())) {
            teamCountInput.setError("Enter number of teams");
            return;
        }

        if (TextUtils.isEmpty(participantsInput.getText())) {
            participantsInput.setError("Enter participants");
            return;
        }

        int teamCount;
        try {
            teamCount = Integer.parseInt(teamCountInput.getText().toString());
            if (teamCount < 2 || teamCount > 6) {
                teamCountInput.setError("Enter between 2-6 teams");
                return;
            }
        } catch (NumberFormatException e) {
            teamCountInput.setError("Invalid number");
            return;
        }

        String[] participantsArray = participantsInput.getText().toString().split("\n");
        if (participantsArray.length < teamCount) {
            participantsInput.setError("Need at least " + teamCount + " participants");
            return;
        }

        // Convert to list and shuffle
        List<String> participants = new ArrayList<>();
        for (String participant : participantsArray) {
            if (!participant.trim().isEmpty()) {
                participants.add(participant.trim());
            }
        }
        Collections.shuffle(participants);

        // Clear previous teams
        teamsContainer.removeAllViews();
        generatedTeams.clear();
        resultsTitle.setVisibility(View.VISIBLE);

        // Create balanced teams
        for (int i = 0; i < teamCount; i++) {
            List<String> team = new ArrayList<>();
            generatedTeams.add(team);
        }

        // Distribute participants using round-robin
        for (int i = 0; i < participants.size(); i++) {
            generatedTeams.get(i % teamCount).add(participants.get(i));
        }

        // Display teams
        for (int i = 0; i < generatedTeams.size(); i++) {
            MaterialCardView teamCard = createTeamCard(i, generatedTeams.get(i));
            teamsContainer.addView(teamCard);
        }

        // Show action buttons
        saveTeamsButton.setVisibility(View.VISIBLE);
        shuffleAgainButton.setVisibility(View.VISIBLE);
    }

    private MaterialCardView createTeamCard(int teamIndex, List<String> teamMembers) {
        MaterialCardView card = (MaterialCardView) getLayoutInflater()
                .inflate(R.layout.team_card, teamsContainer, false);

        TextView teamName = card.findViewById(R.id.team_name);
        teamName.setText("Team " + (teamIndex + 1));

        ImageView teamIcon = card.findViewById(R.id.team_icon);
        teamIcon.setImageResource(getTeamIcon(teamIndex));

        LinearLayout participantsLayout = card.findViewById(R.id.participants_layout);
        for (String member : teamMembers) {
            addParticipantToTeam(participantsLayout, member);
        }

        // Set team color
        card.setCardBackgroundColor(ContextCompat.getColor(this, getTeamColor(teamIndex)));

        return card;
    }

    private void addParticipantToTeam(LinearLayout layout, String participant) {
        TextView participantView = new TextView(this);
        participantView.setText(participant);
        participantView.setTextSize(16);
        participantView.setTextColor(ContextCompat.getColor(this, R.color.white));
        participantView.setTypeface(getResources().getFont(R.font.poppins_regular));
        participantView.setPadding(dpToPx(16), dpToPx(8), dpToPx(16), dpToPx(8));
        layout.addView(participantView);
    }

    private void shuffleTeams() {
        if (generatedTeams.isEmpty()) {
            Toast.makeText(this, "Generate teams first", Toast.LENGTH_SHORT).show();
            return;
        }

        // Combine all participants
        List<String> allParticipants = new ArrayList<>();
        for (List<String> team : generatedTeams) {
            allParticipants.addAll(team);
        }
        Collections.shuffle(allParticipants);

        // Redistribute
        for (List<String> team : generatedTeams) {
            team.clear();
        }

        for (int i = 0; i < allParticipants.size(); i++) {
            generatedTeams.get(i % generatedTeams.size()).add(allParticipants.get(i));
        }

        // Refresh display
        teamsContainer.removeAllViews();
        for (int i = 0; i < generatedTeams.size(); i++) {
            MaterialCardView teamCard = createTeamCard(i, generatedTeams.get(i));
            teamsContainer.addView(teamCard);
        }
    }

    private void saveTeams() {
        if (generatedTeams.isEmpty()) {
            Toast.makeText(this, "No teams to save", Toast.LENGTH_SHORT).show();
            return;
        }

        // In a real app, save to database or shared preferences
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < generatedTeams.size(); i++) {
            sb.append("Team ").append(i + 1).append(":\n");
            for (String member : generatedTeams.get(i)) {
                sb.append("- ").append(member).append("\n");
            }
            sb.append("\n");
        }

        Toast.makeText(this, "Teams saved to history", Toast.LENGTH_LONG).show();
    }

    private int getTeamIcon(int index) {
        int[] teamIcons = {
                R.drawable.ic_team_red,
                R.drawable.ic_team_blue,
                R.drawable.ic_team_green,
                R.drawable.ic_team_yellow,
                R.drawable.ic_team_purple,
                R.drawable.ic_team_orange
        };
        return teamIcons[index % teamIcons.length];
    }

    private int getTeamColor(int index) {
        int[] teamColors = {
                R.color.team_red,
                R.color.team_blue,
                R.color.team_green,
                R.color.team_yellow,
                R.color.team_purple,
                R.color.team_orange
        };
        return teamColors[index % teamColors.length];
    }

    private int dpToPx(int dp) {
        return (int) (dp * getResources().getDisplayMetrics().density);
    }
}

File: app/src/main/java/com/android/ichooseyou/activity/TeamPickerActivity.kt
package com.android.ichooseyou.activity

import android.annotation.SuppressLint
import android.os.Bundle
import android.text.TextUtils
import android.view.View
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.core.content.ContextCompat
import androidx.core.content.res.ResourcesCompat
import com.android.ichooseyou2.R
import com.google.android.material.button.MaterialButton
import com.google.android.material.card.MaterialCardView
import com.google.android.material.textfield.TextInputEditText
import java.util.Collections

class TeamPickerActivity : AppCompatActivity() {
    private lateinit var teamCountInput: TextInputEditText
    private lateinit var participantsInput: TextInputEditText
    private lateinit var generateTeamsButton: MaterialButton
    private lateinit var saveTeamsButton: MaterialButton
    private lateinit var shuffleAgainButton: MaterialButton
    private lateinit var teamsContainer: LinearLayout
    private lateinit var resultsTitle: TextView
    private var itemsList: ArrayList<String> = ArrayList()
    private val generatedTeams: MutableList<MutableList<String>> = ArrayList()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_team_picker)

        // Initialize views
        teamCountInput = findViewById(R.id.team_count_input)
        participantsInput = findViewById(R.id.participants_input)
        generateTeamsButton = findViewById(R.id.generate_teams_button)
        saveTeamsButton = findViewById(R.id.save_teams_button)
        shuffleAgainButton = findViewById(R.id.shuffle_again_button)
        teamsContainer = findViewById(R.id.teams_container)
        resultsTitle = findViewById(R.id.results_title)

        // Get the items list from the intent - Fixed type mismatch
        val intentItems = intent.getStringArrayListExtra("ITEMS")
        if (intentItems != null) {
            itemsList = ArrayList(intentItems)
        }

        // If items were passed, populate the participants field
        if (itemsList.isNotEmpty()) {
            participantsInput.setText(TextUtils.join("\n", itemsList))
        }

        generateTeamsButton.setOnClickListener { generateBalancedTeams() }
        saveTeamsButton.setOnClickListener { saveTeams() }
        shuffleAgainButton.setOnClickListener { shuffleTeams() }
    }

    private fun generateBalancedTeams() {
        // Validate inputs
        if (TextUtils.isEmpty(teamCountInput.text)) {
            teamCountInput.error = "Enter number of teams"
            return
        }
        if (TextUtils.isEmpty(participantsInput.text)) {
            participantsInput.error = "Enter participants"
            return
        }

        val teamCount: Int
        try {
            teamCount = teamCountInput.text.toString().toInt()
            if (teamCount < 2 || teamCount > 6) {
                teamCountInput.error = "Enter between 2-6 teams"
                return
            }
        } catch (e: NumberFormatException) {
            teamCountInput.error = "Invalid number"
            return
        }

        val participantsArray = participantsInput.text.toString().split("\n".toRegex())
            .dropLastWhile { it.isEmpty() }
            .toTypedArray()

        if (participantsArray.size < teamCount) {
            participantsInput.error = "Need at least $teamCount participants"
            return
        }

        // Convert to list and shuffle
        val participants: MutableList<String> = ArrayList()
        for (participant in participantsArray) {
            if (participant.trim().isNotEmpty()) {
                participants.add(participant.trim())
            }
        }
        participants.shuffle()

        // Clear previous teams
        teamsContainer.removeAllViews()
        generatedTeams.clear()
        resultsTitle.visibility = View.VISIBLE

        // Create balanced teams
        for (i in 0 until teamCount) {
            val team: MutableList<String> = ArrayList()
            generatedTeams.add(team)
        }

        // Distribute participants using round-robin
        for (i in participants.indices) {
            generatedTeams[i % teamCount].add(participants[i])
        }

        // Display teams
        for (i in generatedTeams.indices) {
            val teamCard = createTeamCard(i, generatedTeams[i])
            teamsContainer.addView(teamCard)
        }

        // Show action buttons
        saveTeamsButton.visibility = View.VISIBLE
        shuffleAgainButton.visibility = View.VISIBLE
    }

    @SuppressLint("SetTextI18n")
    private fun createTeamCard(teamIndex: Int, teamMembers: List<String>): MaterialCardView {
        val card = layoutInflater
            .inflate(R.layout.team_card, teamsContainer, false) as MaterialCardView
        val teamName = card.findViewById<TextView>(R.id.team_name)
        teamName.text = "Team " + (teamIndex + 1)
        val teamIcon = card.findViewById<ImageView>(R.id.team_icon)
        teamIcon.setImageResource(getTeamIcon(teamIndex))
        val participantsLayout = card.findViewById<LinearLayout>(R.id.participants_layout)
        for (member in teamMembers) {
            addParticipantToTeam(participantsLayout, member)
        }

        // Set team color
        card.setCardBackgroundColor(ContextCompat.getColor(this, getTeamColor(teamIndex)))
        return card
    }

    private fun addParticipantToTeam(layout: LinearLayout, participant: String) {
        val participantView = TextView(this)
        participantView.text = participant
        participantView.textSize = 16f
        participantView.setTextColor(ContextCompat.getColor(this, R.color.white))

        // Fixed API level issue by using ResourcesCompat instead of direct getFont
        ResourcesCompat.getFont(this, R.font.poppins_regular)?.let {
            participantView.typeface = it
        }

        participantView.setPadding(dpToPx(16), dpToPx(8), dpToPx(16), dpToPx(8))
        layout.addView(participantView)
    }

    private fun shuffleTeams() {
        if (generatedTeams.isEmpty()) {
            Toast.makeText(this, "Generate teams first", Toast.LENGTH_SHORT).show()
            return
        }

        // Combine all participants
        val allParticipants: MutableList<String> = ArrayList()
        for (team in generatedTeams) {
            allParticipants.addAll(team)
        }
        allParticipants.shuffle()

        // Redistribute
        for (team in generatedTeams) {
            team.clear()
        }
        for (i in allParticipants.indices) {
            generatedTeams[i % generatedTeams.size].add(allParticipants[i])
        }

        // Refresh display
        teamsContainer.removeAllViews()
        for (i in generatedTeams.indices) {
            val teamCard = createTeamCard(i, generatedTeams[i])
            teamsContainer.addView(teamCard)
        }
    }

    private fun saveTeams() {
        if (generatedTeams.isEmpty()) {
            Toast.makeText(this, "No teams to save", Toast.LENGTH_SHORT).show()
            return
        }

        // In a real app, save to database or shared preferences
        val sb = StringBuilder()
        for (i in generatedTeams.indices) {
            sb.append("Team ").append(i + 1).append(":\n")
            for (member in generatedTeams[i]) {
                sb.append("- ").append(member).append("\n")
            }
            sb.append("\n")
        }
        Toast.makeText(this, "Teams saved to history", Toast.LENGTH_LONG).show()
    }

    private fun getTeamIcon(index: Int): Int {
        val teamIcons = intArrayOf(
            R.drawable.ic_team_red,
            R.drawable.ic_team_blue,
            R.drawable.ic_team_green,
            R.drawable.ic_team_yellow,
            R.drawable.ic_team_purple,
            R.drawable.ic_team_orange
        )
        return teamIcons[index % teamIcons.size]
    }

    private fun getTeamColor(index: Int): Int {
        val teamColors = intArrayOf(
            R.color.team_red,
            R.color.team_blue,
            R.color.team_green,
            R.color.team_yellow,
            R.color.team_purple,
            R.color.team_orange
        )
        return teamColors[index % teamColors.size]
    }

    private fun dpToPx(dp: Int): Int {
        return (dp * resources.displayMetrics.density).toInt()
    }
}